python/python/lance/dataset.py

```diff
@@ -689,6 +689,83 @@ def cleanup_old_versions(
             td_to_micros(older_than), delete_unverified
         )
 
+    def create_scalar_index(
+        self,
+        column: Union[str, List[str]],
+        index_type: str,
+        name: Optional[str] = None,
+        replace: bool = True,
+    ) -> LanceDataset:
+        """Create scalar index on column.
+
```

Comment:
It might be worth explaining what a "scalar" column is, for the lay user. Also mention the index types (right now BTREE) and what they are able to handle (equality and range queries). That way they can know what kind of predicates could benefit from the index. Maybe also worth calling our explicitly that this speeds up ANN search *and* scans, and that it can be used in combination with ANN index.

Reply:
I added a lot more documentation on the function.

Reply:
This is excellent. Thanks for adding this.